packages/wasm/src/view-server.ts

```diff
@@ -73,22 +80,13 @@ export class ViewServer implements ViewServerInterface {
   // As blocks are scanned, the internal wasmViewServer tree is being updated.
   // Flush updates clears the state and returns all the updates since the last checkpoint.
   flushUpdates(): ScanBlockResult {
-    const raw = this.wasmViewServer.flush_updates() as JsonValue;
-    const { height, sct_updates, new_notes, new_swaps } = raw as {
-      height?: string | number | bigint | undefined;
-      sct_updates?: JsonObject | undefined;
-      new_notes?: JsonValue[] | undefined;
-      new_swaps?: JsonValue[] | undefined;
-    };
-    const wasmJson = {
-      new_notes: (new_notes ?? []).map(n => JSON.stringify(n)),
-      new_swaps: (new_swaps ?? []).map(s => JSON.stringify(s)),
-    };
+    const result = this.wasmViewServer.flush_updates() as FlushResult;
+    const { height, sct_updates, new_notes, new_swaps } = result;
     return {
-      height: BigInt(height ?? false),
+      height: BigInt(height ?? 0),
       sctUpdates: validateSchema(SctUpdatesSchema, sct_updates),
-      newNotes: wasmJson.new_notes.map(n => SpendableNoteRecord.fromJsonString(n)),
-      newSwaps: wasmJson.new_swaps.map(s => SwapRecord.fromJsonString(s)),
```

Comment:
The deserialization here felt a bit odd. Given it's so different, it feels like it was intentional, but it's hard to tell without any kind of documentation alongside.

Regardless, it feels unnecessary to do `json -> string -> class` when we can do `json -> class`.

Reply:
i remember this happened for a reason, but i don't remember what that reason was. worth being careful about this.

notably this is one of the few places where a pb type is emitted to js, which apparently works for some reason.

Reply:
ah, iirc something like the `case` statement was different, and perhaps breaking here. but that might have been due to a direct cast

Reply:
I've not run into any errors with the simplified way. We should proceed and re-visit if we later discover if there is an issue.